examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,638 @@
+
+    best_val_epoch = np.argmax(np.array(epoch_val_APs))
+    best_train_epoch = np.argmax(np.array(epoch_train_APs))
+    best_val_ap = epoch_val_APs[best_val_epoch]
+    best_val_test_ap = epoch_test_APs[best_val_epoch]
+    best_val_train_ap = epoch_train_APs[best_val_epoch]
+    best_train_ap = epoch_train_APs[best_train_epoch]
+
+    print("Best Train AP: {:.4f}".format(best_train_ap))
+    print("Best Val AP: {:.4f}".format(best_val_ap))
+    print("Test AP of Best Val: {:.4f}".format(best_val_test_ap))
+    print("Train AP of Best Val: {:.4f}".format(best_val_train_ap))
+    print("TOTAL TIME TAKEN: {:.4f}s".format(time.time() - t0))
+    print("AVG TIME PER EPOCH: {:.4f}s".format(np.mean(per_epoch_time)))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--gpu_id', default=0, help="Please give a value for gpu id")
+    parser.add_argument('--dataset', default="ogbg-molpcba", help="Please give a value for dataset name")
+    parser.add_argument('--seed', default=41, help="Please give a value for seed")
+    parser.add_argument('--epochs', default=450, help="Please give a value for epochs")
+    parser.add_argument('--batch_size', default=2048, help="Please give a value for batch_size")
+    parser.add_argument('--init_lr', default=0.0008, help="Please give a value for init_lr")
+    parser.add_argument('--lr_reduce_factor', default=0.8, help="Please give a value for lr_reduce_factor")
+    parser.add_argument('--lr_schedule_patience', default=8, help="Please give a value for lr_schedule_patience")
+    parser.add_argument('--min_lr', default=0.00001, help="Please give a value for min_lr")
+    parser.add_argument('--weight_decay', default=1e-5, help="Please give a value for weight_decay")
+    parser.add_argument('--L', default=4, help="Please give a value for L")
+    parser.add_argument('--hidden_dim', default=420, help="Please give a value for hidden_dim")
+    parser.add_argument('--out_dim', default=420, help="Please give a value for out_dim")
+    parser.add_argument('--in_feat_dropout', default=0.0, help="Please give a value for in_feat_dropout")
+    parser.add_argument('--dropout', default=0.2, help="Please give a value for dropout")
+    parser.add_argument('--graph_norm', default='True', help="Please give a value for graph_norm")
+    parser.add_argument('--batch_norm', default='True', help="Please give a value for batch_norm")
+
+    # dgn params
+    parser.add_argument('--aggregators', default="mean sum max dir1-dx", type=str, help='Aggregators to use.')
+    parser.add_argument('--root_folder', type=str, default='./')
+
+    args = parser.parse_args()
+
+    # device
+    if args.gpu_id is not None:
+        os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
+        os.environ["CUDA_VISIBLE_DEVICES"] = str(args.gpu_id)
+
+        if torch.cuda.is_available():
+            print('cuda available with GPU:', torch.cuda.get_device_name(0))
+            device = torch.device("cuda")
```

Comment:
Perhaps it will be simpler if you simply do `torch.device("cuda: {}".format(args.gpu_id))`